crates/extensions/c8y_firmware_manager/src/actor.rs

```diff
@@ -733,52 +701,107 @@ impl FirmwareManagerActor {
     // Candidate to be removed since another actor should be in charge of this.
     async fn get_pending_operations_from_cloud(&mut self) -> Result<(), FirmwareManagementError> {
         let message = MqttMessage::new(&C8yTopic::SmartRestResponse.to_topic()?, "500");
-        self.message_box.mqtt_publisher.send(message).await?;
+        self.mqtt_publisher.send(message).await?;
         Ok(())
     }
 }
 
-pub struct FirmwareManagerMessageBox {
-    input_receiver: LoggingReceiver<FirmwareInput>,
+struct FirmwareManagerWorker {
```

Comment:
```suggestion
/// A worker, spawned per request, that can handle a single firmware update request for a child device end-to-end
struct FirmwareManagerWorker {
```
Just trying to highlight that the scope of this worker is per request.

Reply:
Done with a final commit: https://github.com/thin-edge/thin-edge.io/pull/2600/commits/21864b4a57faec913bffe250c8e71a04f03d71ca